server/src/main/java/org/elasticsearch/cluster/routing/allocation/WriteLoadConstraintSettings.java

```diff
@@ -113,7 +115,11 @@ public boolean disabled() {
     private volatile TimeValue queueLatencyThreshold;
 
     public WriteLoadConstraintSettings(ClusterSettings clusterSettings) {
-        clusterSettings.initializeAndWatch(WRITE_LOAD_DECIDER_ENABLED_SETTING, status -> this.writeLoadDeciderStatus = status);
+        if (WRITE_LOAD_DECIDER_ENABLED_FF.isEnabled()) {
+            clusterSettings.initializeAndWatch(WRITE_LOAD_DECIDER_ENABLED_SETTING, status -> this.writeLoadDeciderStatus = status);
+        } else {
+            writeLoadDeciderStatus = WriteLoadDeciderStatus.DISABLED;
+        }
```

Comment:
I think this means we cannot test by overriding the setting in a serverless QA enviroment. Is this intentional? Or maybe we should set the default accordingly based on the feature flag and leave the dynamic update always enabled?